packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -89,25 +87,11 @@ export function createOrUpgradeDb(
     p = p.next(() => upgradeMutationBatchSchemaAndMigrateData(db, txn));
   }
 
-  if (fromVersion < 3 && toVersion >= 3) {
+  if (fromVersion < 4 && toVersion >= 4) {
```

Comment:
Hrm. So it doesn't have to be in this PR, but I would be a fan of having a single block per migration, and then the fromVersion==0 special case will also end up looking more like the above migration.  i.e.:

```
if (fromVersion < 4 && toVersion >= 4) {
  // Brand new clients don't need to rewrite the mutations.
  if (fromVersion !== 0) {
    // mutation queue migration
  }

  // create client metadata store and remote document changes store
}
```

Reply:
I'm gonna pretend this is part of the merge and take care of it in this PR. It does clean this up a bit.